src/db/service/ExecutionService.ts

```diff
@@ -1,11 +1,145 @@
 import { Repository } from 'typeorm';
+import { plainToInstance } from 'class-transformer';
+import { ExecutionModelType } from '../models/ExecutionModel';
 import AppDataSource from '../../data-source';
 import ExecutionEntity from '../entity/ExecutionEntity';
+import { RuleModelType } from '../models/RuleModel';
+import Procedures from '../Procedures';
+import {
+  RowTestType,
+  TableTestType,
+  UnitTestType,
+} from '../models/UnitTestModels';
+import {
+  UnitTestOperations,
+  TableGenericOperations,
+  RecordMatches,
+} from '../entity/enum';
+import { store } from '../redux/store';
+import RowTestService from './RowTestService';
+import TableTestService from './TableTestService';
 
 export default class ExecutionService {
   repository: Repository<ExecutionEntity>;
 
   constructor() {
     this.repository = AppDataSource.getRepository(ExecutionEntity);
   }
+
+  public async checkPassFail(
+    procedure: string,
+    parameters: string[],
+    test: ExecutionModelType
```

Comment:
@alex-hobby
The front end will need to pass this `ExecutionModelType` to the back end so that's the format needed for your form.